app/scripts/minmax/dimMinMax.controller.js

```diff
@@ -283,27 +262,26 @@
 
                     processed_count++;
                     if((processed_count%5000) == 0) {
-                        // If active gaurdian is changed then stop processing combinations
-                        if(vm.active != activeGaurdian) {
+                        // If active gaurdian or page is changed then stop processing combinations
+                        if(vm.active !== activeGaurdian || $location.path() !== '/best') {
                             return;
                         }
-                        $scope.$apply(function () {
-                            vm.progress = processed_count/combos;
-                        });
-                        setTimeout(function() { step(activeGaurdian, h,g,c,l,ci,gh,ar,processed_count); },0);
+                        vm.progress = processed_count/combos;
+                        $timeout(step, 0, true, activeGaurdian, h,g,c,l,ci,gh,ar,processed_count);
                         return;
                     }
                 } ar = 0; } gh = 0; } ci = 0; } l = 0; } c = 0; } g = 0; }
-
+                
                 var tiers = _.each(_.groupBy(Object.keys(set_map), function(set) {
-                    return _.reduce(set.split('/'), function(memo, num){ return memo + parseInt(num); }, 0);
+                    return _.reduce(set.split('/'), function(memo, num){ 
+                        return memo + parseInt(num); }, 0);;
                 }), function(tier) {
                     tier.sort().reverse();
                 });
 
                 vm.allSetTiers = [];
                 var tier_keys = Object.keys(tiers);
-                for (var t = tier_keys.length; t-- > tier_keys.length - 3; ) {
```

Comment:
I had this `- 3` in here so that we only just would see the top 3 tiers a character had, T12, T11, T10... or T10, T9, T8

Just because that dropdown can get to be really large. Also, there probably isn't really a reason someone would want to build a T8 set if they have all the options for a T11 or T10?

Can keep em all in there though, whichever way is fine in the end.